conda/base/context.py

```diff
@@ -942,6 +942,18 @@ def custom_channels(self):
 
     @property
     def channels(self):
+        from ..core.index import check_allowlist
+        from ..models.channel import Channel
+
+        def _validate_denylist_channels(channel_strings: tuple[str, ...]) -> None:
+            """Validate to see if any channels appear in denylist_channels"""
+            channel_urls = tuple(
+                chain.from_iterable(
+                    Channel(channel).base_urls for channel in channel_strings
+                )
+            )
```

Comment:
Can we shove all of this into `check_allowlist` instead?

Reply:
Yes, we actually don't need to bother with this conversion because `Channel` is flexible enough to accept either a URL or just a channel name.